src/utils/uri.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @module "ethpm/utils/uri"
+ */
+
+import { URL } from "url";
+
+function parseScheme(url: URL) {
+  const scheme = url.protocol.replace(/:+$/, "")
+  if(scheme != 'erc1319'){
+	throw new Error(
+	  `Invalid scheme: ${scheme} found on uri: ${url}. Expected scheme to match "erc1319".`
+	)
+  } else {
+	return scheme
+  }
+}
+
+function parseChainId(url: URL) {
+  // todo: support multiple chain ids
+  const chainId = +url.port;
+  if(chainId != 1){
+	throw new Error(
+	  `Invalid chain ID: ${chainId}. Currently only mainnet is supported.`
+	)
+  } else {
+	return chainId
+  }
+}
+
+function parsePackageId(url: URL) {
```

Comment:
these functions can probably just be static methods on the class https://www.typescriptlang.org/docs/handbook/classes.html#static-properties